mitosheet/mitosheet/code_chunks/code_chunk.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+if TYPE_CHECKING:
+    from mitosheet.state import State
+else:
+    State = Any
+
+class CodeChunk:
+    """
+    A CodeChunk is the a abstract base class that can be inhereited from
+    to create a code generating object. For example, a AddColumnCodeChunk
+    with the correct parameters and states will generate code that adds
+    a column with a specific column header to a specific dataframe.
+
+    When a step is transpiled, it generates a list of code chunks. In turn,
```

Comment:
TODO: Understand why it would generate a step would generate a list of code chunks instead of just a single code chunk per step. And then an analysis would generate a list of code chunks

Reply:
Oh, I see. In the case of changing a column dtype for example, we generate the change column dtype code chunk, but then we also need to generate the code chunks for all of the columns that are dependant on it